docs/conventions/component.md

```diff
@@ -0,0 +1,461 @@
+---
+title: Components coding conventions
+label: Components
+---
+
+## General rules
+
+Building components in a project with several systems means that first step in the construction of the component is the base of the component.
+
+In high-level way of thinking, the base should serve as "an interface" and systems' components as specific implementations. Inheritance, though, should be achieved by composition and not by polymorphism.
+
+That's why the base should also be thought of, and managed as, a common layer of re-usable set of rules and guidelines.
+
+All components should follow the same file structure and naming conventions.
+
+If you are not familiar with the concept of systems in ECL, first have a look at [the system documentation](../systems).
+
+### Naming of components
+
+* Throughout the whole project, in `package.json` files, namespace should always be `@ecl/`, regardless whether work is to be done on a generic or system component. One reason is that `@ecl` namespace should be preserved for correct organization on [npmjs](https://www.npmjs.com/).
+* Flavored components should be prefixed with the name of their corresponding system and type (component, style, template): `[generic|ec|eu]-[component|style|template|utility]`.
+
+Example:
+```
+// Some full component name (in package.json)
+@ecl/ec-component-logo
+@ecl/generic-component-navigation-menu
+@ecl/eu-utility-flex
+
+// Corresponding component name (used for file name later)
+ec-component-logo
+generic-component-navigation-menu
+eu-utility-flex
+```
+
+## SCSS
+
+Apart from [rules that should be applied to all scss files](scss), there are some specifications for components:
+
+### File name
+
+SCSS file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.scss
+```
+
+### Generic component
+
+All css rules should be put in mixins.
+
+It could be split in several small mixins, but in any cases we should have a main mixin calling the small ones.
+This main mixin should be named following the component's main class.
```

Comment:
Might worth making a brief mention of that the mixins are usually in `_mixins.scss`. Also, when mentioning the main scss file, could you please give some details about its implication to `package.json`'s `style` and `scss` attributes? I honestly don't know how they related, only guess...